python/cudf/cudf/core/multiindex.py

```diff
@@ -784,6 +775,83 @@ def from_tuples(cls, tuples, names=None):
         result = cls.from_pandas(pdi)
         return result
 
+    def to_arrow(self):
+        """
+        Convert to a PyArrow Table.
+
+        Examples
+        --------
+        >>> import cudf
+        >>> midx = cudf.MultiIndex(
+        ...         levels=[[1, 3, 4, 5], [1, 2, 5]],
+        ...         codes=[[0, 0, 1, 2, 3], [0, 2, 1, 1, 0]],
+        ...         names=["x", "y"],
+        ...     )
+        >>> midx.to_arrow()
+        pyarrow.Table
+        x: int64
+        y: int64
+        """
+        return self._source_data.to_arrow()
```

Comment:
Conceptually treating a MultiIndex as a PyArrow Table is a bit weird from my perspective. You could argue it should be represented as a Struct column instead for example. What needs MultiIndex to support `to_arrow`?

Reply:
This support was added solely because `to_arrow` was broken since `Index.to_arrow` was being triggered incase of `MultiIndex.to_arrow`, there has been no request for this 😄 . Should we not support this and throw an error?

Reply:
Yea I'd be more comfortable throwing here at least for now.

Reply:
Done 👍